package/samplers/auto_sampler/sampler.py

```diff
@@ -0,0 +1,243 @@
+                constant_liar=True,
+            )
+            return
+
+        complete_trials = study.get_trials(deepcopy=False, states=(TrialState.COMPLETE,))
+        complete_trials.sort(key=lambda trial: trial.datetime_complete)
+        if len(complete_trials) >= self._N_COMPLETE_TRIALS_FOR_NSGA:
+            nsga_sampler_cls = (
+                NSGAIISampler
+                if len(study.directions) < THRESHOLD_OF_MANY_OBJECTIVES
+                else NSGAIIISampler
+            )
+            # Use NSGA-II/III if len(complete_trials) <= _N_COMPLETE_TRIALS_FOR_NSGA.
+            self._sampler = nsga_sampler_cls(constraints_func=self._constraints_func, seed=seed)
+
+    def _determine_single_objective_sampler(
+        self, study: Study, trial: FrozenTrial, search_space: dict[str, BaseDistribution]
+    ) -> None:
+        if isinstance(self._sampler, TPESampler):
+            return
+
+        seed = self._rng.rng.randint(MAXINT32)
+        if (
+            self._constraints_func is not None
+            or any(isinstance(d, CategoricalDistribution) for d in search_space.values())
+            or self._include_conditional_param(study)
+        ):
+            # NOTE(nabenabe): The statement above is always true for Trial#1.
+            # Use ``TPESampler`` if search space includes conditional or categorical parameters.
+            # TBD: group=True?
+            self._sampler = TPESampler(
+                seed=seed,
+                multivariate=True,
+                warn_independent_sampling=False,
+                constraints_func=self._constraints_func,
+                constant_liar=True,
+            )
+            return
+
+        complete_trials = study.get_trials(deepcopy=False, states=(TrialState.COMPLETE,))
+        complete_trials.sort(key=lambda trial: trial.datetime_complete)
+        if len(complete_trials) < self._N_COMPLETE_TRIALS_FOR_CMAES:
+            # Use ``GPSampler`` if search space is numerical and
+            # len(complete_trials) < _N_COMPLETE_TRIALS_FOR_CMAES.
+            if not isinstance(self._sampler, GPSampler):
+                self._sampler = GPSampler(seed=seed)
+            return
+
+        if not isinstance(self._sampler, CmaEsSampler):
+            # Use ``CmaEsSampler`` if search space is numerical and
+            # len(complete_trials) > _N_COMPLETE_TRIALS_FOR_CMAES.
+            # Warm start CMA-ES with the first _N_COMPLETE_TRIALS_FOR_CMAES complete trials.
+            warm_start_trials = complete_trials[: self._N_COMPLETE_TRIALS_FOR_CMAES]
+            # NOTE(nabenabe): ``CmaEsSampler`` internally falls back to ``RandomSampler`` for
+            # 1D problems.
+            self._sampler = CmaEsSampler(
+                seed=seed, source_trials=warm_start_trials, warn_independent_sampling=False
+            )
+
+    def _determine_sampler(
+        self, study: Study, trial: FrozenTrial, search_space: dict[str, BaseDistribution]
+    ) -> None:
```

Comment:
How about returning the sampler class on `_determine_sampler`, `_determine_single_objective_sampler`, and `_determine_multi_objective_sampler`? I think many `self._sampler = ...` statements are verbose.

Reply:
Thank you for the suggestion! I addressed it:(